### ISSUE TITLE
The deployed Dokka HTML is empty

### ISSUE THREAD

**Issue by CommanderTvis** [breandan/kotlingrad#26]
The deployed documentation site is empty (I'll try to investigate this on my own, just reporting).

http://breandan.net/kotlingrad/

**Comment by CommanderTvis** [comment]
Seems to be related to https://github.com/Kotlin/dokka/issues/2024

**Comment by breandan** [comment]
Yeah, there is something broken about the Dokka build right now -- I was unable to diagnose the issue, despite having wrestled with Gradle and Dokka for many hours. I'm not actually sure if this is Dokka or Gradle's fault. To be honest, despite having used Gradle for almost a decade, it's a mystery to me how anything works at all. As far as I can tell it's like TeX -- there is no reason, just Google, trial and error.

**Comment by CommanderTvis** [comment]
I quite agree with you avoid Gradle... I personally understand around 30 classes from the Gradle core.

**Comment by breandan** [comment]
I am going to be honest because I really do enjoy many things about Kotlin. By far the most painful part of working in Kotlin today is the build system. When Hans pitched the Kotlin team about integrating @gradle ~8 years ago, I thought it was a mistake then and I am even more convinced now. If Kotlin really is interested in broader adoption and language usability, someone needs to make some hard choices to simplify or rethink their build process. It's incomprehensible. I don't have any more constructive feedback, but they might consider taking some ideas from OCaml's module system or Rust Crates.
